### PR TITLE
Fix Various Scoreboard Build Warnings

### PR DESCRIPTION
**Pull request by simpsont-oci**

### CODE REVIEW

**Review by mitza-oci — APPROVED** [review]

File: dds/idl/value_writer_generator.cpp [hunk 1]
```diff
@@ -71,6 +71,14 @@ namespace {
       return "boolean";
     case AST_PredefinedType::PT_octet:
       return "byte";
+    case AST_PredefinedType::PT_any:
+    case AST_PredefinedType::PT_object:
+    case AST_PredefinedType::PT_value:
+    case AST_PredefinedType::PT_abstract:
+    case AST_PredefinedType::PT_void:
+    case AST_PredefinedType::PT_pseudo:
```
Comment (mitza-oci) [inline dds/idl/value_writer_generator.cpp:79]: Are these lines needed now that we have "default"?

Comment (simpsont-oci) [reply dds/idl/value_writer_generator.cpp:79]: Maybe not? I don't have a build that complains about them in front of me, just the warning message.

File: dds/DCPS/LinuxNetworkConfigMonitor.cpp [hunk 2]
```diff
@@ -50,7 +50,7 @@ bool LinuxNetworkConfigMonitor::open()
   struct {
     nlmsghdr header;
     rtgenmsg msg;
-  } request = {};
+  } request = { {}, {} };
```
Comment (iguessthislldo) [inline dds/DCPS/LinuxNetworkConfigMonitor.cpp:53]: See #2030

Comment (iguessthislldo) [reply dds/DCPS/LinuxNetworkConfigMonitor.cpp:53]: [This conversion in particular](https://github.com/objectcomputing/OpenDDS/pull/2030/files#r520194627)

Comment (simpsont-oci) [reply dds/DCPS/LinuxNetworkConfigMonitor.cpp:53]: ok, I'll back this out.

File: dds/idl/value_writer_generator.cpp (hunk 1 above)
Comment (simpsont-oci) [reply dds/idl/value_writer_generator.cpp:79]: I'll remove them and see what happens.

File: dds/DCPS/LinuxNetworkConfigMonitor.cpp (hunk 2 above)
Comment (iguessthislldo) [reply dds/DCPS/LinuxNetworkConfigMonitor.cpp:53]: Well now that I think about it, that's for XTypes, but it should probably be brought into master for 3.15...

**Comment by simpsont-oci** [comment]
So should I just port the default init + memset to master?

**Comment by iguessthislldo** [comment]
> So should I just port the default init + memset to master?

Yes, I think so.

**Pull request merged**